WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/Coordinator.py

```diff
@@ -17,10 +18,18 @@ def get_players_opponent(self, player) -> Player:
         # Effect activation requires a target player as an argument, so we calculate this here.
         return self.defender if player is self.attacker else self.attacker
 
-    def next_turn(self) -> None:
+    def next_turn(self) -> Turn:
         current_player = self.turnsQueue.turn()
 
-        used_effects = current_player.use_card()
+        # Why record turn before card usage? See Turn class docs.
+        turn = Turn(self.attacker, self.defender, current_player)
+
+        used_card, used_effects = current_player.use_card()
+
+        turn.record_card_usage(used_card)
+
         for effect in used_effects:
-            effect.activate(current_player, self.get_players_opponent(current_player), self.turnsQueue)
+            effect_impact = effect.activate(current_player, self.get_players_opponent(current_player), self.turnsQueue)
+            turn.record_effect_usage(effect_impact)
 
+        return turn
```

Comment:
Hmm, nie wiem czy to jest sensowne żeby zwracać tutaj cokolwiek i w ogóle przechowywać tutaj logikę tworzenia tych obiektów `Turn`. Bardziej widziałbym tutaj zastosowanie wzorca typu listener, nie wiem co uważasz

Reply:
Faktycznie tak będzie lepiej